sql/src/main/java/io/crate/expression/scalar/SubscriptObjectFunction.java

```diff
@@ -23,40 +23,51 @@
 package io.crate.expression.scalar;
 
 import io.crate.data.Input;
-import io.crate.metadata.BaseFunctionResolver;
 import io.crate.metadata.FunctionIdent;
-import io.crate.metadata.FunctionImplementation;
 import io.crate.metadata.FunctionInfo;
 import io.crate.metadata.Scalar;
 import io.crate.metadata.TransactionContext;
-import io.crate.metadata.functions.params.FuncParams;
-import io.crate.metadata.functions.params.Param;
+import io.crate.types.CollectionType;
 import io.crate.types.DataType;
 import io.crate.types.DataTypes;
 import io.crate.types.ObjectType;
 import io.crate.types.StringType;
+import io.crate.types.UndefinedType;
 
 import java.util.List;
 import java.util.Locale;
 import java.util.Map;
 
 public class SubscriptObjectFunction extends Scalar<Object, Map> {
 
-    public static final String NAME = "subscript_obj";
+    private static final String NAME = "subscript_obj";
     private FunctionInfo info;
 
     public static void register(ScalarFunctionModule module) {
-        module.register(NAME,
-            new BaseFunctionResolver(
-                FuncParams.builder(Param.of(ObjectType.untyped()), Param.of(StringType.INSTANCE)).build()
```

Comment:
Why not change the return type here to `Param.ANY` ? This would reduce the memory overhead (less entries in the function registry) and remove the need for  `allRegisteredTypes`

Reply:
Hm don't get your point. This is all about the returnType of the function and not the arguments.
This ensures that the correct data type is used for the according java type which is used on streaming, encoding etc.